### ISSUE TITLE
Add JNDI server support

### ISSUE THREAD

**Issue by antoinesd** [hammock-project/hammock#25]
JNDI server is mandatory if we want to add feature like #24 and JTA support.

**Comment by johnament** [comment]
I think there's some ways to do the integration without JNDI.  Specifically, overriding the [Bittronix integration in hibernate](https://github.com/hibernate/hibernate-orm/blob/master/hibernate-core/src/main/java/org/hibernate/engine/transaction/jta/platform/internal/BitronixJtaPlatform.java) to not use JNDI to look up the user transaction.  This would be true of other JTA impls out there.  Narayana seems to be notorious for its use of JNDI so I'd be cautious to bring it in.

**Comment by antoinesd** [comment]
You're right, we could do without JNDI, but it may imply that we'll have to support specific integration should we have multiple JPA impl support later.
I have to check, but JPA support datasource addition in persistence.xml only thru JNDI lookup.

**Comment by johnament** [comment]
Also not true.  We can implement `PersistenceUnitInfo` and provide the `DataSource` instance as needed.

**Comment by johnament** [comment]
Antoine and I spoke about this.  We agreed, the best solution is to provide a "simple" JNDI like service for binding beans, probably just named beans, in a look up.
